cmd/node-termination-handler.go

```diff
@@ -87,25 +87,27 @@ func main() {
 	cancelChan := make(chan interruptionevent.InterruptionEvent)
 	defer close(cancelChan)
 
-	monitoringFns := map[string]monitorFunc{}
+	monitoringFns := map[string]interruptionevent.Monitor{}
 	if nthConfig.EnableSpotInterruptionDraining {
-		monitoringFns[spotITN] = interruptionevent.MonitorForSpotITNEvents
+		imdsSpotMonitor := interruptionevent.NewSpotInterruptionMonitor(imds, interruptionChan, cancelChan, nthConfig.NodeName)
+		monitoringFns[spotITN] = imdsSpotMonitor
 	}
 	if nthConfig.EnableScheduledEventDraining {
-		monitoringFns[scheduledMaintenance] = interruptionevent.MonitorForScheduledEvents
+		imdsScheduledEventMonitor := interruptionevent.NewScheduledEventMonitor(imds, interruptionChan, cancelChan, nthConfig.NodeName)
+		monitoringFns[scheduledMaintenance] = imdsScheduledEventMonitor
 	}
 
-	for eventType, fn := range monitoringFns {
-		go func(monitorFn monitorFunc, eventType string) {
-			log.Log().Msgf("Started monitoring for %s events", eventType)
+	for _, fn := range monitoringFns {
+		go func(monitor interruptionevent.Monitor) {
```

Comment:
is `type monitorFunc ...` on line 40 necessary anymore?

Reply:
nope